docs/blockentities/container.md

```diff
@@ -0,0 +1,302 @@
+# Containers
+
+A popular use case of [block entities][blockentity] is to store items of some kind. Some of the most essential [blocks][block] in Minecraft, such as the furnace or the chest, use block entities for this purpose. To store items on something, Minecraft uses `Container`s.
+
+The `Container` interface defines methods such as `#getItem`, `#setItem` and `#removeItem` that can be used to query and update the container. Since it is an interface, it does not actually contain a backing list or other data structure, that is up to the implementing system.
+
+Due to this, `Container`s can not only be implemented on block entities, but any other class as well. Notable examples include entity inventories, as well as common modded [items][item] such as backpacks.
+
+:::info
+NeoForge provides the `ItemStackHandler` class as a replacement for `Container`s in many places. It should be used wherever possible in favor of `Container`, as it allows for cleaner interaction with other `Container`s/`ItemStackHandler`s.
```

Comment:
Items themselves should never implement `Container` or anything similar, there is no way to implement the methods in a sane way on an item.

This should also note that other mods should never be expected to implement `Container` or `WorldlyContainer` on any of their inventory-holding things (mainly entities and BEs) and that disregarding the item handler capability severely limits compatibility